java-checks/src/main/java/org/sonar/java/checks/FieldModifierCheck.java

```diff
@@ -31,10 +32,12 @@
 import org.sonar.plugins.java.api.tree.VariableTree;
 
 import java.util.List;
+import java.util.function.Predicate;
 
 @Rule(key = "S2039")
 public class FieldModifierCheck extends IssuableSubscriptionVisitor {
 
+  private static final Predicate<ModifiersTree> MODIFIERS_TREE_PREDICATE = new VisibleForTestingPredicate().negate();
```

Comment:
I'm not a fan of integration tests. I mean by that to test more then one class at a time. It's just the approach you are using `JavaCheckVerifier`. That way is hard to cover all routes on your code and it's also less strait forward. Decoupling the parts of the logic the way I've written is easier to test.